File: M1/azor-chatdog-js/src/commands/sessionSwitch.ts
/**
 * Session switch command with interactive dropdown
 */

import { listSessions } from '../files/sessionFiles.js';
import { selectFromList } from '../cli/prompt.js';
import { printHelp, printError } from '../cli/console.js';

/**
 * Display a formatted session choice
 */
function formatSessionChoice(session: {
  session_id: string;
  title?: string;
  message_count: number;
  last_modified: Date;
  model: string;
}): string {
  const title = session.title || session.session_id;
  const date = session.last_modified.toLocaleString('pl-PL', {
    year: 'numeric',
    month: '2-digit',
    day: '2-digit',
    hour: '2-digit',
    minute: '2-digit',
  });
  return `${title} (${session.message_count} msg, ${date}, ${session.model})`;
}

/**
 * Show interactive dropdown to select a session
 * @returns Selected session ID or null if cancelled
 */
export async function selectSessionInteractive(): Promise<string | null> {
  const sessions = listSessions();

  if (sessions.length === 0) {
    printHelp('\nBrak zapisanych sesji.');
    return null;
  }

  if (sessions.length === 1) {
    printError('Tylko jedna sesja dostępna. Nie można przełączyć.');
    return null;
  }

  // Create choices for inquirer
  const choices = sessions.map((session) => ({
    name: formatSessionChoice(session),
    value: session.session_id,
  }));

  try {
    const selectedId = await selectFromList<string>(
      'Wybierz sesję do przełączenia:',
      choices
    );
    return selectedId;
  } catch (error) {
    // User cancelled (Ctrl+C)
    return null;
  }
}

File: M1/azor-chatdog-py/src/commands/session_switch.py
"""
Interactive session switcher using prompt_toolkit.
"""

from prompt_toolkit.shortcuts import radiolist_dialog
from files import session_files
from cli import console


def select_session_interactive():
    """
    Display an interactive dropdown to select a session.

    Returns:
        str: Selected session ID or None if cancelled
    """
    sessions = session_files.list_sessions()

    if not sessions:
        console.print_help("\nBrak zapisanych sesji.")
        return None

    if len(sessions) == 1:
        console.print_error("Tylko jedna sesja dostępna. Nie można przełączyć.")
        return None

    # Create choices for radiolist_dialog
    choices = []
    for session in sessions:
        if session.get('error'):
            # Skip sessions with errors
            continue

        # Format display text
        title = session.get('title')
        session_id = session['id']
        messages_count = session['messages_count']
        last_activity = session['last_activity']

        if title:
            # Show title with short ID
            short_id = session_id[:8]
            display_text = f"{title} ({messages_count} msg, {last_activity}, ID: {short_id}...)"
        else:
            # Show full ID if no title
            display_text = f"{session_id} ({messages_count} msg, {last_activity})"

        choices.append((session_id, display_text))

    if not choices:
        console.print_error("Brak dostępnych sesji do przełączenia.")
        return None

    try:
        # Show interactive dialog
        result = radiolist_dialog(
            title="Wybierz sesję",
            text="Użyj strzałek ↑↓ do nawigacji, SPACE aby zaznaczyć, ENTER aby potwierdzić:",
            values=choices
        ).run()

        return result
    except (KeyboardInterrupt, EOFError):
        # User cancelled
        return None
